### ISSUE TITLE
i can't get refresh token

### ISSUE THREAD

**Issue by Tahirhan** [googleads/googleads-python-lib#345]
I have OAuth2 Client ID and Secret

when i run generate_refresh_token.py , the file executes and i get a link and it waits for code input from me, but when i click on the link i get this error:

400. That’s an error.

Error: redirect_uri_mismatch

The redirect URI in the request, urn:ietf:wg:oauth:2.0:oob, can only be used by a Client ID for native application. It is not allowed for the WEB client type. You can create a Client ID for native application at https://console.developers.google.com/apis/credentials/oauthc

**Comment by msaniscalchi** [comment]
Hello,

Thanks for the report! You're seeing this issue because the [generate_refresh_token.py](https://github.com/googleads/googleads-python-lib/blob/master/examples/adwords/authentication/generate_refresh_token.py) example expect [Installed Application](https://developers.google.com/identity/protocols/OAuth2#installed) credentials, and the credentials you're using were generated for a 
[Web Application](https://developers.google.com/identity/protocols/OAuth2#webserver) instead.

One of the differences between the two flows is that the [installed application specifies a redirect URI](https://developers.google.com/identity/protocols/OAuth2InstalledApp#request-parameter-redirect_uri) such as `urn:ietf:wg:oauth:2.0:oob`; whereas with the [web flow, you need to provide a authorized redirect URI](https://developers.google.com/identity/protocols/OAuth2WebServer#request-parameter-redirect_uri) where you will complete the remainder of the OAuth 2.0 flow.

In short, you should find your project in the [Google Cloud Console](https://cloud.google.com/console) and either get existing installed application credentials, or [create new credentials](https://developers.google.com/adwords/api/docs/guides/authentication#installed).

Regards,
Mark

**Comment by Tahirhan** [comment]
thank you, i solved the problem. this link : https://github.com/googleads/googleads-python-lib/wiki/API-access-using-own-credentials-(installed-application-flow)#step-2---setting-up-the-client-library helped a lot.
